kyo-core/shared/src/main/scala/kyo/aborts.scala

```diff
@@ -4,95 +4,90 @@ import kyo.core.*
 import scala.reflect.ClassTag
 import scala.util.control.NonFatal
 
-class Aborts[-V] extends Effect[Aborts[V]]:
-    opaque type Command[T] >: Left[V, Nothing] = Left[V, Nothing]
+// Can't use an opaque type because
+// kyo-direct tests crash the compiler.
+type Aborts[-V] //= DoAbort
 
 object Aborts:
-    private case object aborts extends Aborts[Any]
-    def apply[V]: Aborts[V] = aborts.asInstanceOf[Aborts[V]]
 
-    def fail[V](value: V)(using t: Tag[Aborts[V]]): Nothing < Aborts[V] =
-        Aborts[V].fail(value)
+    import internal.*
 
-    def when[V](b: Boolean)(value: => V)(using Tag[Aborts[V]]): Unit < Aborts[V] =
+    def fail[V](v: V): Nothing < Aborts[V] =
+        DoAbort.suspend(v).asInstanceOf[Nothing < Aborts[V]]
+
+    def when[V](b: Boolean)(value: => V): Unit < Aborts[V] =
         if b then fail(value)
         else ()
 
-    def get[V, T](e: Either[V, T])(using Tag[Aborts[V]]): T < Aborts[V] =
-        Aborts[V].get(e)
-
-    extension [V](self: Aborts[V])
-
-        def fail[T <: V](value: T)(using t: Tag[Aborts[T]]): Nothing < Aborts[T] =
-            self.suspend[Nothing](Left(value))
+    def get[V, T](e: Either[V, T]): T < Aborts[V] =
+        e match
+            case Right(v) => v
+            case Left(v)  => DoAbort.suspend(v).asInstanceOf[Nothing < Aborts[V]]
 
-        def when(b: Boolean)(value: => V)(using Tag[Aborts[V]]): Unit < Aborts[V] =
-            if b then fail(value)
-            else ()
+    class RunDsl[V]():
```

Comment:
I'm not a big fan of this approach with DSL classes but I couldn't find a better way. There's a new feature for clause interleaving in Scala 3 that might enable defining a single method but it's still experimental. Suggestions are appreciated!

Reply:
minor:
- unnecessary parenthesis on the class
- Would prefer the name `RunPartiallyApplied` as it can more clearly communicate why this class is like this


I think in Scala 3 you can supply partial type parameter lists. Can you test that? The `V` should come first so it can be specified without the others. Either way I don't think this class is a big deal.

Reply:
After testing locally, I think partial application is best. Do we need to use a dummy+AnyVal to avoid allocations?

```
    class RunPartiallyApplied[V](private val dummy: Boolean = false) extends AnyVal:
        def apply[T: Flat, S, VS, VR](v: T < (Aborts[VS] & S))(
            using
            h: HasAborts[V, VS] { type Remainder = VR },
            ct: ClassTag[V]
        ): Either[V, T] < (VR & S) =
            Abort.handle(handler)(ct, v).asInstanceOf[Either[V, T] < (VR & S)]
    end RunPartiallyApplied
```